src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java

```diff
@@ -126,6 +135,8 @@ public static void setUpJenkins() throws Exception {
     logRecorder.getLoggers().add(t);
     logRecorder.save();
     t.enable();
+    // but even though we can not capture the logs we want to echo them
+    r.showAgentLogs(agent, loggerRule);
```

Comment:
Could probably rework `showAgentLogs` to mirror messages to the controller with a prefix or whatever.

Reply:
Not sure I understand.  
Do you mean set a formatter here that prepends "$agentName"?

Reply:
> something for JTH

Right.

> outside of the scope here

Well, except insofar as it would be handy to pick up here.

Reply:
the logs already contain the agent name.  e.g.

```
61.145 [test-creds #1] using credential myAppCredentialsId
61.145 [test-creds #1] Fetching changes from the remote Git repository
61.206 [slave0] [id=211]	FINE	o.j.p.g.GitHubAppCredentials$DelegatingGitHubAppCredentials#getPassword: Generating App Installation Token for app ID 54321 on agent
61.207 [id=102]	FINE	o.j.p.g.GitHubAppCredentials$DelegatingGitHubAppCredentials$GetToken#call: Generating App Installation Token for app ID 54321 for agent
61.403 [test-creds #1] Checking out Revision a9cd113a37fe363bc18dc40ac3bb1024bbc62842 (refs/remotes/origin/master)
61.558 [test-creds #1] Commit message: "init"
61.578 [id=181]	FINE	o.j.p.g.GitHubAppCredentials#getToken: Generating App Installation Token for app ID 54321
```

So I'm no longer clear what is being asked here @jglick 

[`showAgentLogs`](https://github.com/jenkinsci/jenkins-test-harness/blob/06f783b0a43e389a2f0cf9909bbd8a32bf0981a0/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L1207-L1209) -> [`publish`](https://github.com/jenkinsci/jenkins-test-harness/blob/06f783b0a43e389a2f0cf9909bbd8a32bf0981a0/src/main/java/org/jvnet/hudson/test/JenkinsRule.java#L1236-L1240)

Reply:
perhaps it is the comment that is missleading.  it is in reference to the comment on the [`LoggerRule`](https://github.com/jenkinsci/github-branch-source-plugin/pull/616/files#diff-03b8ddb1abb4051c64c1e2898e14bf79ff304a1fce3442fa3e789f0af96cdeabR98-R99)

https://github.com/jenkinsci/github-branch-source-plugin/blob/a45d053c9987a98c5054dee51b6faa60b19cdf5e/src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java#L98-L102